rhobs/alerting/data_plane/prometheus.gitops_sync_alerts.yaml

```
@@ -0,0 +1,44 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-gitops-sync-alerting
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+  - name: gitops_sync_alerts
+    interval: 1m
+    rules:
+    - alert: GitOpsApplicationSyncErrors
+      expr: |
+        sum(argocd_app_info{sync_status="Synced", namespace="gitops-service-argocd"})
```

It might be that you can overcome the corner case of no synced apps by replacing this with something like this (I didn't try):
```suggestion
        (
        sum(argocd_app_info{sync_status="Synced", namespace="gitops-service-argocd"})
        or sum(argocd_app_info{namespace="gitops-service-argocd"}) * 0
        )
```

This should replace the missing sum with a time series with the same labels a value of zero.

See a reference here: https://promlabs.com/blog/2023/09/13/dealing-with-missing-time-series-in-prometheus/
and a reference to the `or` operator: https://prometheus.io/docs/prometheus/latest/querying/operators/#logical-set-binary-operators

---

In this case we might get a value of `0` which if fine as it is, but we are dividing this `0` with the result of another query.
We most likely get an error for this division.

---

Naive question here, if we have a division by 0 in any situation, that means there are no apps at all.  So even if there is an error, isn't this a don't care situation, and we don't want alerts at all anyway?